### ISSUE TITLE
Slow Rendering on Nvidia Cuda

### ISSUE THREAD

**Issue by Bakery-Bong** [hacksider/Deep-Live-Cam#730]
Hi everyone, firstly thanks to author for this nice program.

I had a lot of troubles but I started this program, I bought special for this soft rtx3070 8gb gpu, and i have 2.5fps on live and 4-5 sec/frame of rendering with CUDA, I saw a lot of people who encountered this problem, but on youtube review people have real good fps. 

I read all issues on github about that, but no any answer, people say that nothing to do with this, anyway I didn't find a solution.

My specs: 
24gb ram ddr4
i3 10gen
rtx3070 8gb
==================================================================

When I run program, my cpu and gpu load for few moments more then 30-40% of work-load, but after when it start rendering in case of LIVE or just face swap, both run on very low resources consumption:

Gpu about 4-6%, 5/8gb of vram
Cpu about 20-30% of load, it's any real solution for that? Or I need few parallel 4090ti to run it?
RAM 11gb~/21gb

Hope it is any solution and somebody can give some useful info. 
===================================================================

Package                      Version
---------------------------- ---------------
absl-py                      2.1.0
addict                       2.4.0
albucore                     0.0.17
albumentations               1.4.18
annotated-types              0.7.0
astunparse                   1.6.3
basicsr                      1.4.2
beautifulsoup4               4.12.3
certifi                      2024.8.30
charset-normalizer           3.4.0
colorama                     0.4.6
coloredlogs                  15.0.1
contourpy                    1.3.0
customtkinter                5.2.2
cv2_enumerate_cameras        1.1.15
cycler                       0.12.1
Cython                       3.0.11
darkdetect                   0.8.0
easydict                     1.13
eval_type_backport           0.2.0
facexlib                     0.3.0
filelock                     3.16.1
filterpy                     1.4.5
flatbuffers                  24.3.25
fonttools                    4.54.1
future                       1.0.0
gast                         0.6.0
gdown                        5.2.0
gfpgan                       1.3.8
google-pasta                 0.2.0
grpcio                       1.66.2
h5py                         3.12.1
humanfriendly                10.0
idna                         3.10
imageio                      2.35.1
importlib_metadata           8.5.0
insightface                  0.7.3
Jinja2                       3.1.4
joblib                       1.4.2
keras                        3.6.0
kiwisolver                   1.4.7
lazy_loader                  0.4
libclang                     18.1.1
llvmlite                     0.43.0
lmdb                         1.5.1
Markdown                     3.7
markdown-it-py               3.0.0
MarkupSafe                   3.0.1
matplotlib                   3.9.2
mdurl                        0.1.2
ml-dtypes                    0.4.1
mpmath                       1.3.0
namex                        0.0.8
networkx                     3.3
numba                        0.60.0
numpy                        1.26.4
onnx                         1.16.0
onnxruntime-gpu              1.16.3
opencv-python                4.8.1.78
opencv-python-headless       4.10.0.84
opennsfw2                    0.10.2
opt_einsum                   3.4.0
optree                       0.13.0
packaging                    24.1
Pillow                       9.5.0
pip                          21.2.3
platformdirs                 4.3.6
prettytable                  3.11.0
protobuf                     4.23.2
psutil                       5.9.8
pydantic                     2.9.2
pydantic_core                2.23.4
Pygments                     2.18.0
pyparsing                    3.1.4
pyreadline3                  3.5.4
PySocks                      1.7.1
python-dateutil              2.9.0.post0
PyYAML                       6.0.2
requests                     2.32.3
rich                         13.9.2
scikit-image                 0.24.0
scikit-learn                 1.5.2
scipy                        1.14.1
setuptools                   57.4.0
six                          1.16.0
soupsieve                    2.6
sympy                        1.13.3
tb-nightly                   2.19.0a20241009
tensorboard                  2.17.1
tensorboard-data-server      0.7.2
tensorflow                   2.17.0
tensorflow-intel             2.17.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.5.0
threadpoolctl                3.5.0
tifffile                     2024.9.20
tk                           0.1.0
tkinterdnd2                  0.4.2
tomli                        2.0.2
torch                        2.0.1+cu118
torchvision                  0.15.2+cu118
tqdm                         4.66.4
typing_extensions            4.12.2
urllib3                      2.2.3
wcwidth                      0.2.13
Werkzeug                     3.0.4
wheel                        0.44.0
wrapt                        1.16.0
yapf                         0.40.2
zipp                         3.20.2
WARNING: You are using pip version 21.2.3; however, version 24.2 is available.
You should consider upgrading via the 'C:\text\Deep-Live-Cam-main\venv\Scripts\python.exe -m pip install --upgrade pip' command.

**Comment by KRSHH** [comment]
Have you installed these two? -

https://visualstudio.microsoft.com/visual-cpp-build-tools/
https://developer.nvidia.com/cuda-11-8-0-download-archive

**Comment by rotatorotator** [comment]
Are you running on windows?

**Comment by KRSHH** [comment]
> 'C:\text\Deep-Live-Cam-main\venv\Scripts\python.exe -m pip install --upgrade pip'

> Are you running on windows?

Yes.

**Comment by rotatorotator** [comment]
Do you have the face enhancer enabled? I have a 4090 n Ununtu and just get 3-4 fps with it enabled, but like 10 fps without face enhancer.
